transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptor.java

```diff
@@ -593,6 +610,80 @@ public void removeBridgeFilters(IoFilterChain filterChain) {
         }
     }
 
+    // If HTTP's X-FORWARDED-FOR header is set, then use it's value to determine the remote
+    // IP address. If X-FORWARDED-FOR header is NOT set, then use IoSession to determine
+    // the remote IP address.
+    private String getRemoteIpAddress(IoSession session, HttpRequestMessage httpRequest) {
+        String remoteIpAddr = httpRequest.getHeader(HEADER_X_FORWARDED_FOR);
+
+        if (remoteIpAddr == null) {
+            SocketAddress socketAddr = session.getRemoteAddress();
```

Comment:
The logic to get IP address walking down the session tree seems to be common (for e.g it is in proxy handler). Can you create a util method for that ?